asab/application.py

```diff
@@ -602,18 +608,38 @@ def _set_housekeeping_time_from_config(self):
 		# if the app started after the housekeeping time, set it to the next day
 		if now > next_housekeeping_time:
 			next_housekeeping_time += datetime.timedelta(days=1)
-		return next_housekeeping_time
+
+		# compute the time limit for the housekeeping
+		time_delta_limit = config_time_limit - config_house_time
+		if time_delta_limit < datetime.timedelta(hours=0):
+			time_delta_limit += datetime.timedelta(days=1)
+
+		next_time_limit = next_housekeeping_time + time_delta_limit
+
+		# Each time has its id that prevents from accidental executing housekeeping twice.
+		next_housekeeping_id = int(now.strftime("%Y%m%d"))
+
+		return (next_housekeeping_time, next_time_limit, next_housekeeping_id)
 
 	def _on_housekeeping_tick(self, message_type):
-		"""Check if it's time for 'Application.housekeeping!'.
-		If so, publish the message and set housekeeping time for the next day.
 		"""
-		if datetime.datetime.now(datetime.timezone.utc) > self.NextHousekeeping:
-			self.PubSub.publish("Application.housekeeping!")
-			self.NextHousekeeping += datetime.timedelta(days=1)
+		Check if it's time for publishing the 'Application.housekeeping!' message.
+		If so, publish the message and set housekeeping time, the time limit and time id for the next day.
+		"""
+		now = datetime.datetime.now(datetime.timezone.utc)
+		today_id = int(now.strftime("%Y%m%d"))
+
+		if self.HousekeepingTime < now:
+			if now < self.HousekeepingTimeLimit and self.HousekeepingId <= today_id:
```

Comment:
isn't the second part of the condition redundant?
if i understand it correctly, then `self.HousekeepingId <= today_id` is a subset of `now < self.HousekeepingTimeLimit`